install-dev/upgrade/sql/1.7.8.0.sql

```diff
@@ -30,3 +30,9 @@ ALTER TABLE `PREFIX_hook` ADD `active` TINYINT(1) UNSIGNED DEFAULT 1 NOT NULL AF
 ALTER TABLE `PREFIX_orders` ADD COLUMN `note` TEXT AFTER `date_upd`;
 
 ALTER TABLE `PREFIX_currency` CHANGE `numeric_iso_code` `numeric_iso_code` varchar(3) NULL DEFAULT NULL;
+
+ALTER TABLE `PREFIX_product_attribute` ADD `date_add` DATETIME NULL AFTER `available_date`;
+ALTER TABLE `PREFIX_product_attribute` ADD `date_upd` DATETIME NULL AFTER `date_add`;
+
+ALTER TABLE `PREFIX_product_attribute_shop` ADD `date_add` DATETIME NULL AFTER `available_date`;
+ALTER TABLE `PREFIX_product_attribute_shop` ADD `date_upd` DATETIME NULL AFTER `date_add`;
```

Comment:
If values can't be null, you should add a new value to both attributes during the upgrade process :thinking:

Reply:
@PierreRambaud  By default this query will set null.
I cannot put dates by default (for example `Now ()`) because it is wrong.
Imagine the following case:
If in my theme the new variations have a "new" label, after the update all the variations will become new!

Reply:
`0000-00-00 00:00:00` ftw ;)

Reply:
In some cases it can be considered as an invalid date

Reply:
@PululuK jk :) maybe it would be good enough to just copy `date_add` from `PREFIX_product` to those tables, but I'm not 100% sure if having `null` in update is bad, I mean it could be bad if someone will set `$autodate = false` and then update of the `Combination` will fail I guess

Reply:
@kpodemski Yes I see ... but the problem is that the combinations are not necessarily created on the same date as the products. For example if I add a combination in a product just before the upgrade, after the upgrade the combination will have the same add date as the product, which is wrong!

Reply:
@PululuK sure, but this value is useful for developers only after an upgrade, before it we don't have it, I'm expecting that this will help developers sometime after an upgrade where they'll change this value anyway

Reply:
@PululuK no matter what dates you put in those fields in upgrade script they are all will be wrong or unknown (null) 🤷

Reply:
@kpodemski  I think that `NULL` is the best because it reflects the situation: **Before the upgrade the date for these combination did not exist.(unknown)** 

But of course `date_upd` will be updated as soon as it has actions on the combinations.

Reply:
@PululuK 

ok, then, I suggest you to double check safety of using `$combination->update()` with your `null` values, ok? I'm not sure if `ObjectModel` will not fill `date_add` as current date... which is also wrong.

Reply:
@PululuK I'm writing about this part of code: https://github.com/PrestaShop/PrestaShop/blob/develop/classes/ObjectModel.php#L722

but it's late and maybe I'm missing something :)

Reply:
> ok, then, I suggest you to double check safety of using $combination->update() with your null values, ok?

Don't get mad man! It's just a discussion 😅
[this behavior is questionable](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/ObjectModel.php#L722) ... but I see what you mean. I will look at this later. 
Thank you for your patience

Reply:
hey i'm not mad :D we're discussing to make sure that after upgrade all combinations stay as "old" not "new ones" :D

let's just double check this, if it's ok and after update of some old `Combination` `date_add` will stay `null`, I'm approving PR 👍

Reply:
I'll check it later ... Thanks @kpodemski

Reply:
The declaration and the update are not the same.
In the declaration, you tell MySQL those fields are `NOT NULL`, but during the upgrade' script fields can be `NULL`.

The best way to do it is by allowing fields to be null.

Reply:
Or as @jolelievre said, we can use the Product date_add and date_upd values during the upgrade

Reply:
Thanks guys !
See https://github.com/PrestaShop/PrestaShop/pull/21521#issuecomment-714803046